rules/private/copy_file_private.bzl

```diff
@@ -111,27 +114,29 @@ def copy_file(name, src, out, is_executable = False, **kwargs):
       is_executable: A boolean. Whether to make the output file executable. When
           True, the rule's output can be executed using `bazel run` and can be
           in the srcs of binary and test rules that require executable sources.
+          WARNING: If `allow_symlink` is True, `src` must also be executable.
+      allow_symlink: A boolean. Whether to allow symlinking instead of copying.
+          When False, the output is always a hard copy. When True, the output
+          *can* be a symlink, but there is no guarantee that a symlink is
+          created (i.e., at the time of writing, we don't create symlinks on
+          Windows). Set this to True if you need fast copying and your tools can
+          handle symlinks (which most UNIX tools can).
       **kwargs: further keyword arguments, e.g. `visibility`
     """
+
+    copy_file_impl = _copy_file
     if is_executable:
-        _copy_xfile(
-            name = name,
-            src = src,
-            out = out,
-            is_windows = select({
-                "@bazel_tools//src/conditions:host_windows": True,
-                "//conditions:default": False,
-            }),
-            **kwargs
-        )
-    else:
-        _copy_file(
-            name = name,
-            src = src,
-            out = out,
-            is_windows = select({
-                "@bazel_tools//src/conditions:host_windows": True,
-                "//conditions:default": False,
-            }),
-            **kwargs
-        )
+        copy_file_impl = _copy_xfile
+
+    copy_file_impl(
+        name = name,
+        src = src,
+        out = out,
+        is_windows = select({
+            "@bazel_tools//src/conditions:host_windows": True,
+            "//conditions:default": False,
+        }),
+        is_executable = is_executable,
```

Comment:
If this were     executable = is_executable  would you just be able to call _copy_file() and not have _copy_xfile()?

Reply:
AFAICT, no. `_copy_xfile = rule(executable = True)` has a different (native) parent rule than `_copy_file = rule(executable = False)` (https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/analysis/skylark/StarlarkRuleClassFunctions.java;l=290;drc=8b856f5484f0117b2aebc302f849c2a15f273310).